core/database/foxx/tests/user_router.test.js

```diff
@@ -0,0 +1,130 @@
+        type: 4,  // transfer token
+        other_token_data: "1cbaaee5-b938-4a4e-87a8-f1ec4d5d92f9",   // fake UUID
+    };
+
+    it("should accept a valid user's token and execute an update", () => {
+        // arrange
+        // NOTE: the get request has query params instead of a body
+        const query_params = test_params[0];
+        // TODO: make encoded query params less hard coded
+        const request_string = `${usr_base_url}/token/set?client=${query_params._key}&access=${query_params.access}&refresh=${query_params.refresh}&expires_in=500000`;
+
+        // act
+        const response = request.get(request_string);
+
+        // assert
+        expect(response.status).to.equal(204);
+
+        const user_token_data = db.u.document({_key: query_params._key});
+        expect(user_token_data).to.include(query_params);
+    });
+
+    it("should update only a valid user's token when additionally provided type", () => {
+        // arrange
+        const local_test_params = test_params[1];
+        const query_params = {
+          ...local_test_params,
+          type: 4,
+        };
+        // TODO: make encoded query params less hard coded
+        const request_string = `${usr_base_url}/token/set?client=${query_params._key}&access=${query_params.access}&refresh=${query_params.refresh}&expires_in=500000&type=${query_params.type}`;
+
+        // act
+        const response = request.get(request_string);
+
+        // assert
+        expect(response.status).to.equal(204);
+
+        const user_token_data = db.u.document({_key: query_params._key});
+        expect(user_token_data).to.include(local_test_params);
+
+        const user_token_edge = db.globus_token.outEdges(user_token_data._id);
+        expect(user_token_edge).to.be.empty;
+    });
+
+    it("should update only a valid user's token when additionally provided other token data", () => {
+        // arrange
+        const lcoal_test_params = test_params[2];
+        const query_params = {
+            ...lcoal_test_params,
+            other_token_data: "1cbaaee5-b938-4a4e-87a8-f1ec4d5d92f9",
+        };
+        // TODO: make encoded query params less hard coded
+        const request_string = `${usr_base_url}/token/set?client=${query_params._key}&access=${query_params.access}&refresh=${query_params.refresh}&expires_in=500000&other_token_data=${query_params.other_token_data}`;
+
+        // act
+        const response = request.get(request_string);
+
+        // assert
+        expect(response.status).to.equal(204);
+
+        const user_token_data = db.u.document({_key: query_params._key});
+        expect(user_token_data).to.include(lcoal_test_params);
+
+        const user_token_edge = db.globus_token.outEdges(user_token_data._id);
+        expect(user_token_edge).to.be.empty;
+    });
+
+    it("should accept a valid user's token with additional type and globus collection data and add it to globus_token edge", () => {
+        // arrange
```

Comment:
This test is great!

I would also recommend adding tests that throw errors if non valid other_token_data is provided.